juju/application.py

```diff
@@ -41,14 +41,14 @@ def status(self):
         """Get the application status, as set by the charm's leader.
 
         """
-        return self.data['status']['current']
+        return self.__getattr__('status')['current']
```

Comment:
There was a naming conflict with `self.status` in one of the methods (which also caused an issue with using `getattr()`), and the others included hyphenation.  I'll make something cleaner because I agree that it sucks.